templates/CRM/Contact/Page/DedupeFind.tpl

```diff
@@ -84,6 +190,206 @@
 {include file='CRM/common/dedupe.tpl'}
 {literal}
 <script type="text/javascript">
-var oTable = null;
+CRM.$(function($) {
+  var sourceUrl = {/literal}'{$sourceUrl}'{literal};
+  var context   = {/literal}'{$context}'{literal};
+  $('#dupePairs').DataTable({
+    //"scrollX": true, // doesn't work with hover popup for for icons
+    "lengthMenu": [[10, 25, 50, 100, 1000, 2000, -1], [10, 25, 50, 100, 1000, 2000, "All"]],
+    "searching" : true,
+    "dom": 'flrtip',
+    "processing": true,
+    "serverSide": true,
+    "ajax": sourceUrl,
+    "columns"  : [
+      {data: "is_selected_input"},
+      {data: "src_image"},
+      {data: "src"},
+      {data: "dst_image"},
+      {data: "dst"},
+      {data: "src_email"},
+      {data: "dst_email"},
+      {data: "src_street"},
+      {data: "dst_street"},
+      {data: "src_postcode"},
+      {data: "dst_postcode"},
+      {
+        data: "conflicts",
+        className: "crm-pair-conflict"
+      },
+      {data: "weight"},
+      {data: "actions"},
+    ],
+    "order": [], // removes default order by for column 1 (checkbox)
+    "columnDefs": [ 
+      {
+        "targets": [0, 1, 3, 13],
+        "orderable": false
+      },
+      {
+        "targets": [7, 8, 9, 10, 11, 12],
+        "visible": false
+      }
+    ],
+    rowCallback: function (row, data) {
+      // Set the checked state of the checkbox in the table
+      $('input.crm-dedupe-select', row).prop('checked', data.is_selected == 1);
+      if (data.is_selected == 1) {
+        $(row).toggleClass('crm-row-selected');
+      }
+      // for action column at the last, set nowrap
+      $('td:last', row).attr('nowrap','nowrap');
+      // for conflcts column
+      $('td.crm-pair-conflict', row).attr('nowrap','nowrap');
+    }
+  });
```

Comment:
Since this is a function it can't be added to the markup as data, but you could add the data via javascript:
    $('#dupePairs').data('table', {
      rowCallback: function...
   });

Just be sure you do this outside of a document.ready callback so it runs first before the datatable gets initialized by Common.js